test/lib/sdkdriver/src/transport/websocket/client.ts

```diff
@@ -0,0 +1,100 @@
+import Transport from '../';
+import { EventEmitter } from 'events';
+import * as WebSocket from 'ws';
+
+/**
+ * WebSocket client {@link Transport}.
+ * @class
+ * @fires Transport#close
+ * @fires Transport#message
+ */
+export default class WSClientTransport extends EventEmitter implements Transport {
+  private readonly  _deps: any;
+  private readonly _sendBuffer: Array<string>;
+  private readonly _wsUrl: string;
+  private _wsClient: WebSocket | null;
+
+  /**
+   * Constructor.
+   * @param {string} wsUrl - WebSocket URL
+   */
+  constructor(wsUrl: string, deps: any = {}) {
+    super();
+    this._deps = {
+      WebSocket,
+      ...deps
+    };
+    this._sendBuffer = [];
+    this._wsClient = null;
+    this._wsUrl = wsUrl;
+  }
+
+  /**
+   * Close the {@link WSClientTransport}.
+   * @returns {void}
+   */
+  close(): void {
+    if (!this._wsClient) {
+      return;
+    }
+    try {
+      this._wsClient.close();
+    } catch (e) {
+      // Do nothing.
+    }
+    this._wsClient = null;
+  }
+
+  /**
+   * Open the {@link WSClientTransport}.
+   * @returns {Promise<void>}
+   */
+  open(): Promise<void> {
+    const { WebSocket } = this._deps;
+    if (!this._wsClient) {
+      this._wsClient = new WebSocket(this._wsUrl);
+    }
+    const wsClient: WebSocket = this._wsClient as WebSocket;
```

Comment:
writing this without the `if` statement may allow omitting the `as WebSocket` cast:

```ts
const wsClient = this._wsClient || new WebSocket(this._wsUrl);
this._wsClient = wsClient;
```